src/libraries/Common/tests/System/IO/Compression/CompressionStreamUnitTestBase.cs

```diff
@@ -494,6 +496,69 @@ async Task<long> GetLengthAsync(CompressionLevel compressionLevel)
             Assert.True(fastestLength >= optimalLength);
             Assert.True(optimalLength >= smallestLength);
         }
+
+        [Theory]
+        [MemberData(nameof(ZLibOptionsRoundTripTestData))]
+        public async Task RoundTripWithZLibCompressionOptions(string testFile, ZLibCompressionOptions options)
+        {
+            using var uncompressedStream = await LocalMemoryStream.readAppFileAsync(testFile);
+            var compressedStream = CompressTestFile(uncompressedStream, options);
+            using var decompressor = CreateStream(compressedStream, mode: CompressionMode.Decompress);
+            using var decompressorOutput = new MemoryStream();
+            await decompressor.CopyToAsync(decompressorOutput);
+            decompressor.Dispose();
```

Comment:
The changes in this file are mixing async and sync code, and each has a different path that needs to be tested separately.

If `CopyToAsync` is being used, then let's:
- Create the compressor instance with `await using` instead of just `using`
- Call `DisposeAsync` instead of `Dispose`
- Call `FlushAsync` instead of `Flush`

And we should have similar tests that do not call any async code: Use `CopyTo`, `Dispose`, `Flush`, and `using` to create the compressor instance.

Reply:
> And we should have similar tests that do not call any async code: Use CopyTo, Dispose, Flush, and using to create the compressor instance.

Yep, we should use async for all, I will update that, though why we need the same test with non async version? The test is for checking if compressing with various options would produce different result after decompressing.

Reply:
OK, maybe it's not needed if we are not adding separate paths for sync or async.